app/controllers/server.rb

```diff
@@ -22,7 +20,19 @@ class Server < Sinatra::Base
 
     post '/sources/:IDENTIFIER/data' do
       p1 = Payload.payload_constructor(params)
-      binding.pry
+      client = Client.find_by(identifier: params[:IDENTIFIER])
+      if !client
+        status 403
+        body "Client doesn't exist."
+      elsif Payload.already_exists?(params, client.id)
+        status 403
+        body "This is already entered"
+      elsif p1.values.include?(nil) || parsed_payload.empty?
```

Comment:
Hi @bfpepper ! We were curious as to what parsed_payload is? We don't see this anywhere else.